src/WebJobs.Extensions.Http/Extensions/HttpRequestExtensions.cs

```diff
@@ -42,5 +48,33 @@ public static async Task<string> ReadAsStringAsync(this HttpRequest request)
             // last one wins for any duplicate query parameters
             return request.Query.ToDictionary(p => p.Key, p => p.Value.Last());
         }
+
+        public static ClaimsIdentity GetAppServiceIdentity(this HttpRequest request)
+        {
+            if (!request.Headers.ContainsKey(EasyAuthIdentityHeader))
+            {
+                return null;
+            }
+            string headerValue = request.Headers[EasyAuthIdentityHeader].First();
+            return FromBase64EncodedJson(headerValue);
+        }
+      
+        private static ClaimsIdentity FromBase64EncodedJson(string payload)
+        {
+            using (var buffer = new MemoryStream(Convert.FromBase64String(payload)))
+            {
+                ClaimsIdentitySlim slim = (ClaimsIdentitySlim)Serializer.ReadObject(buffer);
+                return slim.ToClaimsIdentity();
+            }
+        }
+
+        private static DataContractJsonSerializer GetCustomSerializer()
```

Comment:
Since this serializer is specific to ClaimsIdentitySlim, recommend renaming this method GetClaimsIdentitySerializer and also renaming your static property from Serializer to ClaimsIdentitySerializer.